test/stdlib/CharacterProperties.swift

```diff
@@ -142,7 +142,14 @@ CharacterPropertiesTests.test("Casing") {
   expectTrue(Character("π").isLowercase)
 
   expectEqual("SS", Character("ß").uppercased())
+
+// Some versions of ICU on Linux (62.1) have a bug producing the wrong value
+// when lowercasing "ẞ". Darwin platforms never shipped this version, so
+// conditionally test based on platform.
+#if _runtime(_ObjC)
```

Comment:
Linux could in theory have the ObjC runtime. Can we conditionalize this on "Darwin"?

Reply:
The convention for asking "is this a Darwin platform that has Apple's implementation of the objc runtime and Foundation" is spelled `#if _runtime(_ObjC)`. This is an internal conditional check (not exposed to users) and should return false on any Linux, where interop would otherwise be broken.

This predates me and I don't really know why it's done this way, but it's followed consistently throughout the standard library in hundreds of places. CC @jrose-apple and @jckarter who might have opinions about this convention and whether it should change.

Reply:
It well predates `canImport`, which is at least part of the reason.

Reply:
Does `#if _runtime(_ObjC)` check whether the *runtime* is available or merely ObjC interoperability is enabled? I ask because there are a few tests that are run twice, once with interop and once without, and regardless of platform.

Reply:
It's asking if interop is enabled, i.e. "will the Swift runtime be one that interoperates with Objective-C".

Reply:
@milseman – Can we please change this to "#ifdef darwin"? I'd really like to see the conflation of the ObjC runtime with the Darwin platform kept to a minimum.

Reply:
We have `-enable-objc-interop` which controls whether ObjC interop is present or not.  Darwin and ObjC are not synonymous.

Reply:
For all practical purposes, they are. I would bet that any theoretical interop with any other ObjC runtime and Foundation implementation is going to require a lot more than a binary flag here, since the existing code paths for generating metadata and integrating with ObjC runtime and Foundation facilities are extremely specific to Apple's implementation. Maybe we can change the name to `_runtime(_AppleObjC)` and change the flag to `-enable-objc-interop=apple` if it makes you all feel better, but I think this is the right condition to use in the standard library, regardless of what we name it.

Reply:
I filed https://bugs.swift.org/browse/SR-9358 for that.